samples/bluetooth/fast_pair/locator_tag/src/fp_adv.c

```diff
@@ -430,52 +446,53 @@ void app_fp_adv_mode_set(enum app_fp_adv_mode adv_mode)
 	}
 }
 
-void app_fp_adv_rpa_rotation_suspend(bool suspended)
+static void app_fp_adv_request_start(void)
 {
-	fp_adv_rpa_rotation_suspended = suspended;
+	bool has_account_key = bt_fast_pair_has_account_key();
+	bool requested_adv_on = app_fp_adv_trigger_is_enabled_any();
+	enum app_fp_adv_mode requested_mode;
+
+	if (requested_adv_on) {
+		requested_mode = has_account_key ?
+				 APP_FP_ADV_MODE_NOT_DISCOVERABLE :
+				 APP_FP_ADV_MODE_DISCOVERABLE;
+	} else {
+		requested_mode = APP_FP_ADV_MODE_OFF;
+	}
+
+	if (!requested_adv_on && (fp_adv_mode == APP_FP_ADV_MODE_OFF)) {
+		LOG_INF("Fast Pair: advertising: already disabled");
+		return;
+	} else if (requested_adv_on && (fp_adv_mode != APP_FP_ADV_MODE_OFF)) {
+		LOG_INF("Fast Pair: advertising: already enabled, refreshing");
```

Comment:
With the current approach, it would be better to provide an explicit API call to trigger advertising refresh and avoid automatic refresh when a module updates its request bit (to avoid refreshing advertising accidentally).

Reply:
Is refresh needed just to update the advertising payload?